### PR TITLE
chore: deploy post build

### PR DESCRIPTION
**Pull request by mpetrunic**
Alternative to #89 

@kalambet I think this makes more sense for this project as it involves fewer build minutes and less of a wait because you don't have to install and build twice.

### CODE REVIEW

**Comment by kalambet** [comment]
@mpetrunic I thought about it before too, but there were two reasons why I initially dismissed it because: (a) not sure if can preserve updating (writing) repo deployments; (b) I wanted to exclude deployment for the draft PRs but leave the `CI` check. The (b) was not implemented just yet bc I did not check the docs, but tbh same goes with (a) too :)

Do you think it is make sense to preserve deployments and do not deploy to CF Pages on Draft PRs?

**Comment by mpetrunic** [comment]
> @mpetrunic I thought about it before too, but there were two reasons why I initially dismissed it because: (a) not sure if can preserve updating (writing) repo deployments; (b) I wanted to exclude deployment for the draft PRs but leave the `CI` check. The (b) was not implemented just yet bc I did not check the docs, but tbh same goes with (a) too :)
> 
> Do you think it is make sense to preserve deployments and do not deploy to CF Pages on Draft PRs?

Well deployment on its own is "free", and takes very little time 15s. If it's done in a separate job, it's best to cache the build so you avoid reinstalling dependencies and rebuilding.

As for draft, I think you can create if condition that will not execute on draft PRs

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -28,4 +28,12 @@ jobs:
         fi
         echo -e "Found problematic uppercase section links:\n$found"
         exit 1
-    - run: yarn build
\ No newline at end of file
+    - run: yarn build
+    - name: Publish to Cloudflare Pages
+      uses: cloudflare/pages-action@1
+      with:
```
Comment (kalambet) [inline .github/workflows/check.yaml:34]:
```suggestion
      if: '! github.event.pull_request.draft'
      with:
```

Comment (kalambet) [reply .github/workflows/check.yaml:34]: @mpetrunic do you think we also can add
```
     permissions:
        contents: read
        deployments: write
```

Section on the `steps` level so the `GITHUB_TOKEN` can be generated with enough permissions to update the repo deployments?

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

**Pull request merged**